packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmMapInternal.kt

```diff
@@ -127,13 +159,15 @@ internal interface MapOperator<K, V> : CollectionOperator<V, RealmMapPointer> {
     ): Pair<V?, Boolean> {
         realmReference.checkClosed()
         return insertInternal(key, value, updatePolicy, cache)
+            .also { modCount += 1 }
```

Comment:
You can use `modcount++` or `++modcount`. as I pointed it here https://github.com/realm/realm-kotlin/pull/1221#discussion_r1089042339, sorry I didn't follow the conversation 😴.

The documentation says that the `inc`/`dec` must not modify the object but to return a new value. The result of that call would be assigned to the variable on which it was invoked. 

So if `modcount = 0` if we call `modcount++`, the int 0 is not modified, but a new integer 1 is computed and reasigned to modcount.
If instead of integers it were an object, the instance before and after invoking `++` would be different.